inc/ticket.class.php

```diff
@@ -4356,6 +4515,7 @@ function showForm($ID, $options = []) {
             // If entity is not in the list of user's entities,
             // then use as default value the first value of the user's entites list
             $this->fields["entities_id"] = $this->userentities[0];
+            $this->input["entities_id"] = $this->userentities[0];
```

Comment:
IMHO, forcing input is not a good idea.
Unless I am wrong, passing `$options['entities_id']` to `$this->getDefaultContract()` should work.

Reply:
I thought too but it seems like it doesn't change anything in the case.
`$this->input["entities_id"]` will be set directly in an HTML input from the value in `$this->fields["entities_id"]` by this code in `CommonITILObject::showActorsPartForm()`:
```php
if ($this->userentity_oncreate
     && isset($this->countentitiesforuser)
     && ($this->countentitiesforuser > 1)) {
    echo "<br>";
    $rand = Entity::dropdown(['value'     => $this->fields["entities_id"],
                                   'entity'    => $this->userentities,
                                   'on_change' => 'this.form.submit()']);
 } else {
    echo "<input type='hidden' name='entities_id' value='".$this->fields["entities_id"]."'>";
 }
```

Not sure what to think of this, can change if you insist but why shouldn't we set the input ?

Reply:
As a developper, I expect `$this->input` to be set by user input, and eventually by business rules. If some values are forced by a really specific logic, most of developpers will not be aware of this (it may make debugging harder, code may be unvoluntary altered/removed, ...).
As `$options['entities_id']` is already set and is used to adapt proposed values on some fields (proposed actors, default ticket type, ...), it would be logical to use it to adapt default contract type.
Also, `$options['entities_id']` is set to original item entity when a ticket is created from an item.